### PR TITLE
[spec] add service-meshes.proto

### LINKED ISSUE #33: Enum for `service_mesh` type
To guarantee compatibility between SMPS implementations, we may want to use an enum for `service_mesh`.

_Originally posted by @leecalcote in https://github.com/layer5io/service-mesh-performance-specification/pull/31_

### PR DESCRIPTION
**Pull request by leecalcote**
fixes #33 

Should we have a separate file or keep the enum inside the same proto?

### CODE REVIEW

**Comment by leecalcote** [comment]
> Looks great, I think separate file works well. Should we specify the service mesh version also?

Good call, yes. It's not obvious, but [`mesh_build`](https://github.com/layer5io/service-mesh-performance-specification/blob/master/smps.proto#L148) is meant to capture the service mesh's version #. While it's clear that this field needs renamed, it's not clear to me whether we should move it into `service-meshes.proto` or not. WDYT?

**Comment by leecalcote** [comment]
> We will need another make command for service-meshes.proto

Thank you, @kushthedude. I have updated the `makefile` to include this new photo.

**Comment by leecalcote** [comment]
@kushthedude good call. Will do.

**Comment by leecalcote** [comment]
Protos compiled.

File: spec/service-meshes.pb.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
+// Code generated by protoc-gen-go. DO NOT EDIT.
```
Comment (kushthedude) [inline spec/service-meshes.pb.go:1]: This can be removed then I guess @leecalcote ?

Comment (leecalcote) [reply spec/service-meshes.pb.go:1]: That is automatically generated by `protoc` and will be included each time `make build` runs.

Comment (leecalcote) [reply spec/service-meshes.pb.go:1]: I'm not sure that we need to (or really should) have the generated classes stored in this repo.

Comment (kushthedude) [reply spec/service-meshes.pb.go:1]: No actually, this is not the generated one. It was the old file when the rename was not done.

Comment (leecalcote) [reply spec/service-meshes.pb.go:1]: Ooooh.

Comment (kushthedude) [reply spec/service-meshes.pb.go:1]: > I'm not sure that we need to (or really should) have the generated classes stored in this repo.

I second that.

**Comment by leecalcote** [comment]
Thank you, sir.

**Pull request merged**